src/ratbag_emu/protocol/util/profile.py

```diff
@@ -1,47 +1,89 @@
 from collections import namedtuple
 
+
 '''
 Imeplemnts a profile with all the mouse settings
 '''
 class Profile(object):
     Led = namedtuple('Led', ['red', 'green', 'blue'])
 
-    def __init__(self, obj=None):
+    class Button(object):
+        def __init__(action_type, val=None):
+            self.action_type = action_type
+            self.button = None
+            self.special = None
+            self.macro = None
+
+            setattr(self, self.action_type, val)
+
+    def __init__(self, obj):
         '''
         Hardware properties
         '''
-        self.active_dpi = 0
-        self.x_dpi = self.y_dpi = [
-            3000,
-            6000
-        ]
-        self.step = 100
+        self.default_dpi = None
+        self.active_dpi = None
+        self.dpi = []
+        if 'resolutions' in obj:
+            for i, dpi in enumerate(obj['resolutions']):
+                self.dpi.append(dpi['xres'])
+                try:
+                    if dpi['is_active']:
```

Comment:
IIRC @whot left a comment explaining that with json you better use the `get`:
```python
if dpi.get('is_active', False):
    self.active_dpi = i
```
This way you can remove all of the `try/catch KeyError`.

Reply:
Sure.